src/app/frontend/replicasetdetail/replicaset_service.js

```diff
@@ -64,4 +66,41 @@ export class DeleteReplicaSetService {
 
     return deferred.promise;
   }
+
+  /**
+   * Opens an update replica set dialog. Returns a promise that is resolved/rejected when user wants
+   * to update the replicas. Nothing happens when user clicks cancel on the dialog.
+   *
+   * @param {!backendApi.ReplicaSetDetail} replicaSetDetail
+   * @returns {!angular.$q.Promise}
+   */
+  showUpdateReplicasDialog(replicaSetDetail) {
```

Comment:
These two methods (on this service and on the dialog) take too broad arguments. They take entire replica set detail, whereas they only need name, current, and desired numbers of pods. This volatiles law of demeter that says that you should only have direct dependencies that you use in params list. Having broader arguments makes the API harder to use. 
If you change this to have only directly used parameters, the usage in src/app/frontend/replicasetlist/replicasetcardmenu_controller.js will be easy. The test will be cleaner as well. 

I think this wasn't noticed before, because this was called from details page which had details object and everything looked good.